server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStore.java

```diff
@@ -214,7 +214,7 @@ protected void performExpirationClean() {
         Instant last = lastClean.get();
         //check if we should expire again
         Instant now = Instant.now();
-        if (Duration.between(last, now).getNano() > getExpirationTime().getNano()) {
+        if (Duration.between(last, now).getSeconds() > getExpirationTime().getSeconds()) {
```

Comment:
why not using toMillis for both, then we are on same level as before with "currentTimeMillis"

Reply:
ok , so we have other usages of Duration in combination always with toMillis, 

However I did some debugging of the tests and for me there is still the bug that the cleanup is called to often.
I dont see an update of lastClean.set(Instant.now());

See tests cleanUpLegacyCodesCodesWithoutExpiresAtAfter3Days and cleanUpExpiredTokensBasedOnExpiresField
They expect 0 because cleanup is done ALWAYS, but with default 5 min the existing codes stay in DB.

Reply:
see here from my debugging I updated lastCleanup
https://github.com/strehle/uaa/commit/3109aed26ebeee26aa657beaa6f9e1a0cbae0cee